appservice/src/TunnelProxy.ts

```diff
@@ -170,6 +171,28 @@ export class TunnelProxy {
         this._server.unref();
     }
 
+    // Starts up an app when it is found to be in the STOPPED state
+    // Apps can be in the STOPPED state for different reasons:
+    //  1. A stop request was sent through the Azure API (using the portal, using the extension, etc)
+    //    - In this case it will respond with 403 until a start request is sent to the Azure API
+    //  2. The app is inactive, or was recently started
```

Comment:
It feels odd to me to do `await this._client.start();` over and over again. I think I'd prefer to check `this._client.getState()` before `checkTunnelStatusWithRetry` and prompt the user with something like: `The web app must be running, but is currently in state "Stopped".` with an option to start the web app.

Unless `await this._client.start();`is magically fixing the tunnel? It's not clear to me if that or the ping is fixing it.

Reply:
- this._client.start() is addressing the case where the user stopped the app.
- Pinging is addressing the case where the app has gone to sleep (the user did not stop the app in this case)

Both of those scenarios show a tunnel status of STOPPED so I wasn't able to differentiate between the two. Yes it would potentially send this._client.start() even if the app was already started, but that seemed harmless in my testing.

I will see if this._client.getState() gives me a different state than the tunnel status. If I can use that to differentiate between the situations above, then we can make the behavior a bit cleaner. Unfortunately I can't log in to Azure to test at the moment (maybe there is an outage?) but I will report back when I can test.

Reply:
Ok, I verified that this._client.getState() will show me if the app was started or stopped by the user. So:
- getState=Stopped means that the app needs to be started before proceeding
- getState=Running, tunnel status=STOPPED means that pinging the app will get it started